clients/admin-ui/src/pages/login.tsx

```diff
@@ -11,206 +11,223 @@ import {
   Stack,
   useToast,
 } from '@fidesui/react';
-import { useFormik } from 'formik';
+import { Formik } from 'formik';
 import type { NextPage } from 'next';
 import Head from 'next/head';
-import Image from 'next/image';
 import { useRouter } from 'next/router';
-import { signIn } from 'next-auth/react';
-import React, { useState } from 'react';
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { login, selectToken, useLoginMutation } from '../features/auth';
+import Image from '../features/common/Image';
 
 const useLogin = () => {
-  const [isLoading, setIsLoading] = useState(false);
+  const [loginRequest, { isLoading }] = useLoginMutation();
+  const token = useSelector(selectToken);
   const toast = useToast();
   const router = useRouter();
-  const formik = useFormik({
-    initialValues: {
-      email: '',
-      password: '',
-    },
-    onSubmit: async (values) => {
-      setIsLoading(true);
-      const response = await signIn<'credentials'>('credentials', {
-        ...values,
-        callbackUrl: `${window.location.origin}`,
-        redirect: false,
+  const dispatch = useDispatch();
+
+  const initialValues = {
+    email: '',
+    password: '',
+  };
+
+  const onSubmit = async (values: typeof initialValues) => {
+    const credentials = {
+      username: values.email,
+      password: values.password,
+    };
+    try {
+      const user = await loginRequest(credentials).unwrap();
+      dispatch(login(user));
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.log(error);
+      toast({
+        status: 'error',
+        description:
+          'Login failed. Please check your credentials and try again.',
       });
-      setIsLoading(false);
-      if (response && response.ok) {
-        router.push('/');
-        toast.closeAll();
-      } else {
-        toast({
-          status: 'error',
-          description:
-            'Login failed. Please check your credentials and try again.',
-        });
-      }
-    },
-    validate: (values) => {
-      const errors: {
-        email?: string;
-        password?: string;
-      } = {};
+    }
+  };
+
+  const validate = (values: typeof initialValues) => {
+    const errors: {
+      email?: string;
+      password?: string;
+    } = {};
 
-      if (!values.email) {
-        errors.email = 'Required';
-      }
-      /* Disable email validation while client-level authentication is required
-       *
-       * else if (
-       *   !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)
-       * ) {
-       *   errors.email = 'Invalid email address';
-       * }
-       */
+    if (!values.email) {
+      errors.email = 'Required';
+    }
 
-      if (!values.password) {
-        errors.password = 'Required';
-      }
+    if (!values.password) {
+      errors.password = 'Required';
+    }
```

Comment:
a note for the future, I'd love to start using [yup](https://github.com/jquense/yup) for validation

Reply:
@allisonking Is there a benefit to using yup that I'm not seeing? 👀

Reply:
I like that `yup` makes validation declarative (you declare all the requirements a field has to meet at the same time you define the form (i.e. required, minimum 8 letters, etc.) and then it'll do the validation on submit (or I think you can also do it on touch) so then you don't have to do manual validation by looking through all the fields yourself with a bunch of `if`s. since validation is defined in a schema, you can also reuse that schema which is handy. it's also nice that formik supports yup out of the box and they recommend it in their tutorial too: https://formik.org/docs/tutorial#schema-validation-with-yup